storage/Utils/Logger.cc

```diff
@@ -0,0 +1,98 @@
+
+
+#include <iostream>
+
+#include "storage/Utils/Logger.h"
+#include "storage/Utils/AppUtil.h"
+
+
+namespace storage
+{
+
+    static Logger* my_logger = nullptr;
+
+
+    Logger*
+    get_logger()
+    {
+	return my_logger;
+    }
+
+
+    void
+    set_logger(Logger* logger)
+    {
+	my_logger = logger;
+    }
+
+
+    bool
+    Logger::test(LogLevel log_level, const std::string& component)
+    {
+	return true;
+    }
+
+
+    class StdoutLogger : public Logger
+    {
+	virtual void write(LogLevel log_level, const std::string& component, const std::string& file,
+			   int line, const std::string& function, const std::string& content) override;
+    };
+
+
+    void
+    StdoutLogger::write(LogLevel log_level, const std::string& component, const std::string& file,
+			int line, const std::string& function, const std::string& content)
+    {
+	std::cout << datetime(time(NULL), true, true) << " <" << log_level << "> [" << component << "] "
+		  << file << "(" << function << "):" << line << " " << content << std::endl;
+    }
+
+
+    Logger*
+    get_stdout_logger()
+    {
+	static StdoutLogger stdout_logger;
+
+	return &stdout_logger;
+    }
+
+
+    Silencer::Silencer()
+	: active(false)
+    {
+	turn_on();
+    }
+
+
+    Silencer::~Silencer()
+    {
+	turn_off();
+    }
+
+
+    void
+    Silencer::turn_on()
+    {
+	if (active)
+	    return;
+
+	active = true;
+	++count;
+    }
+
+
+    void
+    Silencer::turn_off()
+    {
+	if (!active)
+	    return;
+
+	active = false;
+	--count;
```

Comment:
I do not get logic with counting. I expect turn_off do not set active false if count is bigger then one. Accoring to code it looks like count xor active flag is useless.
I do not get whole logic of it. As count cannot be bigger then one ( as it is guarded by active flag ). So what is purpose of that? Using only active flag ( when you want current behavior that on,on,off result in off ) or count ( when you want counting like on,on,off to be still on till next off ).